aws/src/main/java/org/apache/iceberg/aws/glue/LockManagers.java

```diff
@@ -0,0 +1,241 @@
+
+    InMemoryLockManager(Map<String, String> properties) {
+      initialize(properties);
+    }
+
+    @VisibleForTesting
+    void acquireOnce(String entityId, String ownerId) {
+      DefaultLockContent content = LOCKS.get(entityId);
+      if (content != null && content.expireMs() > System.currentTimeMillis()) {
+        throw new IllegalStateException(String.format("Lock for %s currently held by %s, expiration: %s",
+            entityId, content.ownerId(), content.expireMs()));
+      }
+
+      long expiration = System.currentTimeMillis() + heartbeatTimeoutMs();
+      boolean succeed;
+      if (content == null) {
+        DefaultLockContent previous = LOCKS.putIfAbsent(
+            entityId, new DefaultLockContent(ownerId, expiration));
+        succeed = previous == null;
+      } else {
+        succeed = LOCKS.replace(entityId, content, new DefaultLockContent(ownerId, expiration));
+      }
+
+      if (succeed) {
+        // cleanup old heartbeat
+        if (HEARTBEATS.containsKey(entityId)) {
+          HEARTBEATS.remove(entityId).cancel(false);
+        }
+
+        HEARTBEATS.put(entityId, scheduler().scheduleAtFixedRate(() -> {
+          DefaultLockContent lastContent = LOCKS.get(entityId);
+          try {
+            long newExpiration = System.currentTimeMillis() + heartbeatTimeoutMs();
+            LOCKS.replace(entityId, lastContent, new DefaultLockContent(ownerId, newExpiration));
+          } catch (NullPointerException e) {
+            throw new RuntimeException("Cannot heartbeat to a deleted lock " + entityId, e);
+          }
+
+        }, 0, heartbeatIntervalMs(), TimeUnit.MILLISECONDS));
+
+      } else {
+        throw new IllegalStateException("Unable to acquire lock " + entityId);
+      }
+    }
+
+    @Override
+    public boolean acquire(String entityId, String ownerId) {
+      try {
+        Tasks.foreach(entityId)
+            .retry(Integer.MAX_VALUE - 1)
+            .onlyRetryOn(IllegalStateException.class)
+            .throwFailureWhenFinished()
+            .exponentialBackoff(acquireIntervalMs(), acquireIntervalMs(), acquireTimeoutMs(), 1)
+            .run(id -> acquireOnce(id, ownerId));
+        return true;
+      } catch (IllegalStateException e) {
+        return false;
+      }
+    }
+
+    @Override
+    public void release(String entityId, String ownerId) {
+      DefaultLockContent currentContent = LOCKS.get(entityId);
+      if (currentContent == null) {
+        throw new IllegalArgumentException("Cannot find lock for entity " + entityId);
+      }
+
+      if (!currentContent.ownerId().equals(ownerId)) {
+        throw new IllegalArgumentException(String.format(
+            "Cannot unlock %s by %s, current owner: %s", entityId, ownerId, currentContent.ownerId()));
+      }
+      HEARTBEATS.remove(entityId).cancel(false);
+      LOCKS.remove(entityId);
+    }
+
+    @Override
+    public void close() {
```

Comment:
I think this also needs to shut down the heartbeat thread pool. That should probably be done in the base class, so probably just call `super.close()` here and add the shutdown there.

Reply:
Oh, I see. Thanks, I missed that because `scheduler()` is not a static method.